internal/rpc/starknet/types.go

```diff
@@ -339,20 +333,19 @@ func NewReceipt(receipt types.TxnReceipt) (Receipt, error) {
 }
 
 type CommonReceiptProperties struct {
-	TxnHash          string `json:"txn_hash"`
-	TransactionIndex uint64 `json:"transaction_index"`
-	ActualFee        string `json:"actual_fee"`
-	Status           string `json:"status"`
-	StatusData       string `json:"status_data"`
-	BlockHash        string `json:"block_hash"`
-	BlockNumber      uint64 `json:"block_number"`
+	TransactionHash string `json:"transaction_hash"`
+	ActualFee       string `json:"actual_fee"`
+	Status          string `json:"status"`
+	StatusData      string `json:"status_data"`
+	BlockHash       string `json:"block_hash"`
+	BlockNumber     uint64 `json:"block_number"`
 }
 
 type InvokeTxReceipt struct {
 	CommonReceiptProperties
-	MessagesSent    []*MsgToL1 `json:"messages_sent"`
-	L1OriginMessage *MsgToL2   `json:"l1_origin_message"`
-	Events          []*Event   `json:"events"`
+	MessagesSent    []*MsgToL1 `json:"messages_sent,omitempty"`
+	L1OriginMessage *MsgToL2   `json:"l1_origin_message,omitempty"`
+	Events          []*Event   `json:"events,omitempty,omitempty"`
```

Comment:
Are these fields all options in `InvokeTxReceipt`? I would only omit those which are optional

Reply:
`MessagesSent` and `Events` are required, you are right, and the `L1OriginMessage` is removed on the next spec version. I will fix that

Reply:
Done -> https://github.com/NethermindEth/juno/pull/382/commits/c7d4d3784b4010d884b101ad28c9293c71bdb0e2